docs/build/message-history.md

```diff
@@ -0,0 +1,116 @@
+
+## Run a message history server
+
+To enable the message history feature, you must run your own [**xmtp-message-history-server**](https://github.com/xmtp/xmtp-message-history-server). When you do, it will provide a message history sync URL to use in the next step.
+
+## Set message history sync URL
+
+When your app creates a new client, specify the location where the client should sync message history.
+
+<Tabs groupId="sdk-langs">
+<TabItem value="rn" label="React Native" attributes={{className: "rn_tab"}}>
+
+```jsx
+Code sample coming soon
+```
+
+</TabItem>
+<TabItem value="kotlin" label="Kotlin" attributes={{className: "kotlin_tab"}}>
+
+```kotlin
+Client().create(
+    account = alixWallet,
+    options = ClientOptions(
+        ClientOptions.Api(XMTPEnvironment.PRODUCTION, true),
+        enableV3 = true,
+        appContext = context
+        historySyncUrl = "SYNC_URL"
+    )
+)
+```
+
+</TabItem>
+<TabItem value="swift" label="Swift"  attributes={{className: "swift_tab"}}>
+
+```swift
+Code sample coming soon
+```
+
+</TabItem>
+<TabItem value="node" label="Node"  attributes={{className: "node_tab"}}>
+
+```tsx
+Code sample coming soon
+```
+
+</TabItem>
+</Tabs>
+
+## Request message history sync
+
+When a new client is created, it will look for other clients registered with the same inbox ID. 
+
+If there are none, there is no message history to be synced.
+
+If there are clients registered with the same inbox ID, it will request a message history sync with the clients. 
+
+For example, let’s say that inbox ID 1/client B requests a message history sync with inbox ID 1/client A.
+
+The request encodes, encrypts, and sends inbox ID 1/client A’s group chat messages to the message history server indicated by inbox ID 1/client A’s `historySyncUrl`. 
+
+The request then downloads the encrypted messages from the message history server to inbox ID 1/client B’s local database. Inbox ID 1/client B can then decrypt, decode, and display the messages.
+
+The request performs this sync for every client registered with inbox ID 1.
+
+![Sequence diagram illustrating the request message history sync flow between Inbox 1/Client B, Inbox 1/Client A, and a message history server](img/req-message-hist-sync.png)
+
+:::caution
+Downloading message history may involve a large amount of data. To help users avoid potential mobile data charges, consider prompting users to initiate the download only when connected to Wi-Fi.
+:::
+
+<Tabs groupId="sdk-langs">
+<TabItem value="rn" label="React Native" attributes={{className: "rn_tab"}}>
+
+```jsx
+Code sample coming soon
+```
+
+</TabItem>
+<TabItem value="kotlin" label="Kotlin" attributes={{className: "kotlin_tab"}}>
+
+```kotlin
+client.requestMessageHistorySync()
+```
+
+</TabItem>
+<TabItem value="swift" label="Swift"  attributes={{className: "swift_tab"}}>
+
+```swift
+Code sample coming soon
```

Comment:
How about this? I pointed Cursor AI at Client.swift in xmtp-ios and asked it to give me a code sample to request message history sync:

```suggestion
let client: Client = // ... your initialized XMTP client

do {
    try await client.requestMessageHistorySync()
    print("Message history sync requested successfully")
} catch {
    print("Failed to request message history sync: \(error)")
}
```

Reply:
Looks good!